src/k8s/pkg/k8sd/app/hooks_start.go

```diff
@@ -13,5 +17,12 @@ func (a *App) onStart(s *state.State) error {
 		go a.nodeConfigController.Run(s.Context)
 	}
 
+	// start control plane config controller
+	if a.controlPlaneConfigController != nil {
+		go a.controlPlaneConfigController.Run(s.Context, func(ctx context.Context) (types.ClusterConfig, error) {
+			return utils.GetClusterConfig(ctx, s)
+		})
```

Comment:
go routines -- i haven't learned this in my training yet.  Kinda like python `async` maybe? or more like an async thread?

Reply:
Go routines are green threads (user-space). further, they are not colored separately (e.g. no need to do `func foo()` vs `async func foo()`. Let's discuss offline and go into more details